sample_files/problem_setting/examples/standard_interactor.cpp

```diff
@@ -0,0 +1,179 @@
+#include <cstdio>
+#include <cstdlib>
+#include <fstream>
+#include <regex.h>
+#include <stdexcept>
+#include <string>
+
+namespace CheckerCodes {
+int AC = 0;
+int WA = 1;
+int PARTIAL = 7;
+} // namespace CheckerCodes
+
+void assertWA(bool);
+
+// Implementation of the tricky whitespace logic for standard checkers.
+namespace standard_whitespace_detail {
+enum WhitespaceFlag { NONE = 0, SPACE = 1, NEWLINE = 2, ALL = 3 };
+WhitespaceFlag current_flag = ALL; // At checker start, consume all whitespace.
+
+void pokeFlag(WhitespaceFlag flag) {
+  if (current_flag != NONE && (current_flag != NEWLINE || flag != ALL)) {
+    throw std::runtime_error("Never call two whitespace methods in a row, "
+                             "except for readNewLine() followed by readEOF().");
+  }
+  current_flag = flag;
+}
+
+enum ConsumeResult {
+  NO_WHITESPACE,
+  NO_LINES,
+  LINES,
+};
+ConsumeResult consumeWhitespace() {
```

Comment:
Can we make sure functions are properly separated? Either manually or via clang-format. This is an eyesore.

Reply:
Fixed. The idea was to condense template code for users, but it's probably just worse.